### PR TITLE
(WIP) Internals Monitor

### PR DESCRIPTION
**Pull request by pykello**
Being able to monitor internal data structures of Citus is useful for (1) understanding how Citus works, (2) Debugging.

- [x] `citus_connections_hash()` for monitoring `ConnectionHash` in `connection_management.c`.
- [x] `citus_connection_placements_hash()` for monitoring `ConnectionPlacementHash` in `placement_connection.c`.
- [x] List of queries sent over a connection.
- [x] transaction modification level.

### CODE REVIEW

File: src/backend/distributed/citus--7.4-2--7.4-3.sql [hunk 1]
```diff
@@ -10,4 +10,88 @@ COMMENT ON FUNCTION worker_hash_partition_table(bigint, integer, text, text, oid
                                                 anyarray)
     IS 'hash partition query results';
 
+CREATE FUNCTION citus_connections_hash()
```
Comment (marcocitus) [inline src/backend/distributed/citus--7.4-2--7.4-3.sql]: Could we put these in the citus schema and make sure regular users cannot call them?

File: src/backend/distributed/connection/remote_commands.c [hunk 2]
```diff
@@ -339,11 +340,29 @@ pchomp(const char *in)
...
 {
+	MemoryContext oldContext;
+	RemoteCommandLogRecord *logRecord;
+
 	if (!LogRemoteCommands)
 	{
 		return;
 	}
 
+	oldContext = MemoryContextSwitchTo(TopMemoryContext);
```
Comment (marcocitus) [inline src/backend/distributed/connection/remote_commands.c]: I usually just do:
```
SET client_min_messages TO LOG;
SET citus.log_remote_commands TO on;
```

**Comment by pykello** [comment]
> Hey Hadi, do you think it is worth revisiting this PR? It makes it easier to learn about Citus internals.

I think having this will be useful. At the time I was doing this, my impression was that current data structures in connection managements are a bit overcomplicated and hard to understand, so we might want to simplify them a bit before moving this forward. I'll look at it again and come up with a plan.

**Comment by pykello** [comment]
Now that I have better understanding of Citus internals, I'm closing this in favour of a future plan to create a better internals monitor tool.

**Pull request closed without merging**